### ISSUE TITLE
Keras implementation

### ISSUE THREAD

**Issue by alef07** [Kyubyong/neural_tokenizer#1]
Hi!

I'm trying to reimplement your code in Keras.
I've reimplemented the whole architecture, but I'm facing a problem to deal with your loss function.

You loss function is computed over the logits: 
`   self.loss = tf.nn.sparse_softmax_cross_entropy_with_logits(logits=self.logits, labels=self.y)
`
Then you compute your final loss function only of the non masked elements:
`  self.mean_loss = tf.reduce_sum(self.loss * self.istarget) / (tf.reduce_sum(self.istarget))
`
However, with keras, the output given to the model are 1 - 0 vectors, and not logits:
`model = Model(input=x, outputs=preds)`

Computing the loss with preds (result of the argmax on the logits layer), gives me obviously bad results, because of the sparsity.  It seems the framework does not permit to train on a certain type of data (logits) and to predict another type of data (1 - 0 vectors).

Do you know how I could deal with that with Keras?

Thanks!
Thank you again for the good work ;)

Paul,

**Comment by alef07** [comment]
Ok I found how to deal with this problem :-)
I perform the argmax after the predict( and the behaviour is strictly identical.
THanks anyway.
